coordinate/coordinator.go

```diff
@@ -0,0 +1,301 @@
+	}
+
+	return nil
+
+}
+
+// StartCluster will
+// 1. create the flota dir that all the daemons will register to.
+// 2. wait until it all the daemons have been registered
+// 3. will timeout on its wait and return false or return true if all found
+//
+// These should only be used by the flotilla client
+func (c *Client) StartCluster(numdaemons int, startsleep int) (bool, error) {
+
+	err := c.getOrCreateRoot()
+	if err != nil {
+		c.StopCluster()
+		panic(err)
+	}
+	if err != nil {
+		panic(err)
+	}
+
+	// Wait for the cluster to spin up for the sleep time or return when
+	// the cluster is all up. It checks every second.
+	step := 0
+
+	for {
+
+		step++
+		count := c.ClusterCount()
+		log.Printf("%d daemons have registered in %d seconds", count, step)
+
+		if count >= numdaemons {
+			return true, nil
+		}
+
+		if step > startsleep {
+			for _, member := range c.ClusterMembers() {
+				log.Printf("Member %s had joined", member)
+			}
+			return false, errors.New("Daemons not all registered")
+		}
+
+		time.Sleep(time.Second * 1)
+	}
+
+	return false, errors.New("Unable to start Cluter")
+}
+
+// StopCluster will close all watcher go routines and do a recursive delete on
+// rootDir + flota. This should only be used by the flotilla client
+func (c *Client) StopCluster() {
+	c.Close()
+	resp, err := c.client.Delete(rootDir+c.flota, false)
+	if resp == nil || err != nil {
+		panic(ErrorUnableToStopCluster)
+	}
+}
+
+// ClusterCount will get against the rootdir + the flota and count how many nodes
+// it finds.
+func (c *Client) ClusterCount() int {
+
+	// starting at the root flota node we need to go through all the registered nodes
+	// and pull out their ip address values (which are a comma seperated list)
+	resp, err := c.client.Get(rootDir+c.flota, false, false)
+
+	if err != nil {
+		return 0
+	}
+	if resp.Node.Dir == true {
+		return len(resp.Node.Nodes)
+	}
+
+	return 0
+}
+
+// ClusterMembers will concatanate all deamon's who have registered IP addresses into
+// a []string of IP addresses
+func (c *Client) ClusterMembers() []string {
+
+	var results []string
+
+	// starting at the root flota node we need to go through all the registered nodes
+	// and pull out their ip address values (which are a comma seperated list)
+	resp, err := c.client.Get(rootDir+c.flota, false, false)
+
+	if err != nil {
+		return nil
+	}
+	if resp.Node.Dir == true {
+		for _, node := range resp.Node.Nodes {
+			values := strings.Split(node.Value, ",")
+			results = append(results, values...)
+		}
+	}
+	return results
+}
+
+// Register will add to the path under the rootdir and flota so its IP address
+// can be pulled down.
+// TODO: Expand this to cover ports and maybe a way to pick a single address
+func (c *Client) Register(port int) error {
+
+	// This is to ensure the client started and the test should actually be run
+	resp, cerr := c.client.Get(rootDir+c.flota, false, false)
+	if resp == nil || cerr != nil {
+		return cerr
```

Comment:
You are potentially returning `nil` here. If `resp == nil`, it will short circuit.